### PR TITLE
GH-7397: Handle symlinked Dockerfiles

### LINKED ISSUE #7397: docker-compose up fails with symlinked Dockerfile
## Description of the issue

I have several microservices which share an identical `Dockerfile`. To simplify maintenance and reduce duplication I would like to replace all the Dockerfiles with a symlink to a single "common" `Dockerfile`.

Before symlinking, my directory structure looks like this:

```
- docker-compose.yaml
- microservice_a/
  - Dockerfile
- microservice_b/
  - Dockerfile
```

after, it looks like this:

```
- docker-compose.yaml
- common/
  - Dockerfile
- microservice_a/
  - Dockerfile  ->  ../common/Dockerfile
- microservice_b/
  - Dockerfile  ->  ../common/Dockerfile
```

Without symlinking, `docker-compose up` works as expected.

Once I switch to the symlink model, this no longer works, and I get errors related to `Dockerfile: no such file or directory`. See below for more detail:

## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.25.4, build 8d51620a
```

**Output of `docker version`**
```
Docker version 19.03.8, build afacb8b
```

(ommited docker-compose config because it contains confidential information like private keys which I'm configuring using env vars)

## Steps to reproduce the issue

1. Create two services with idential dockerfiles and reference them both in a compose file.
2. Move their dockerfile to a subdirectory 'common' and symlink it in the services' Dockerfiles.

### Observed result

`docker-compose` can no longer find the Dockerfile, even though it's still in "the same place" because of the symlink.

### Expected result

`docker-compose` continues to function, and can read the Dockerfile through the new symlink.

### Stacktrace / full error message

```
[+] Building 0.0s (2/2) FINISHED
 => [internal] load .dockerignore                                                                                                  0.0s
 => => transferring context: 34B                                                                                                   0.0s
 => [internal] load build definition from Dockerfile                                                                               0.0s
 => => transferring dockerfile: 70B                                                                                                0.0s
failed to solve with frontend dockerfile.v0: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount384971895/common/Dockerfile: no such file or directory
Traceback (most recent call last):
  File "docker-compose", line 6, in <module>
  File "compose/cli/main.py", line 72, in main
  File "compose/cli/main.py", line 128, in perform_command
  File "compose/cli/main.py", line 1077, in up
  File "compose/cli/main.py", line 1073, in up
  File "compose/project.py", line 548, in up
  File "compose/service.py", line 351, in ensure_image_exists
  File "compose/service.py", line 1110, in build
  File "compose/progress_stream.py", line 25, in stream_output
  File "compose/utils.py", line 61, in split_buffer
  File "compose/utils.py", line 37, in stream_as_text
  File "compose/service.py", line 1818, in build
FileNotFoundError: [Errno 2] No such file or directory: '/var/folders/7_/wk5b0_312sq4_9x59jhq0q_r0000gn/T/tmp0jhuoqwn'
[20348] Failed to execute script docker-compose
```

## Additional information

MacOS Catalina 10.15.4

### PR DESCRIPTION
**Pull request by Code0x58**
Resolves #7397

### CODE REVIEW

File: compose/service.py [hunk 1]
```diff
@@ -1090,7 +1103,7 @@ def build(self, no_cache=False, pull=False, force_rm=False, memory=None, build_a
             forcerm=force_rm,
             pull=pull,
             nocache=no_cache,
-            dockerfile=build_opts.get('dockerfile', None),
```
Comment (Code0x58) [inline compose/service.py:1093]: When the path is a URL or gzip is true, `APIClient` passes dockerfile through to the server, so this should probably still be provided in those cases

**Comment by thaJeztah** [comment]
Thanks! Looking at my steps trying to reproduce the problem (see https://github.com/docker/compose/issues/7397#issuecomment-621513714), I'm actually wondering if this should be fixed in docker-compose; I'm thinking this should be fixed in the docker cli (or in buildkit), and docker-compose should pass the `dockerfile` path as-is (leaving the handling to the CLI)

**Comment by Code0x58** [comment]
Nice drill-down! I like the idea of it being cleaned up and handled in the CLI, it sounds like it could touch a lot of complements - I'm certainly interested to see how it's managed.

I'm wondering if this PR would find it's way in as a workaround at the docker-compose level, until changes are made to the CLI (and python library?), then here would need updating again.

**Comment by glours** [comment]
Thanks for taking the time to create this issue/pull request!

Unfortunately, Docker Compose V1 has reached end-of-life and we are not accepting any more changes (except for security issues). Please try and reproduce your issue with Compose V2 or rewrite your pull request to be based on the v2 branch and create a new issue or PR with the relevant Compose V2 information.

**Pull request closed without merging**